feat: map clusters

marcellmueller opened the pull request:
#909
<img width="1456" alt="Screenshot 2025-06-30 at 3 22 19 PM" src="https://github.com/user-attachments/assets/929f9f30-7b1a-4433-9734-b15866ae24f3" />

marcellmueller commented on public/frontend/src/components/search/SearchMap/layers/recreationFeatureLayer.ts:
```diff
@@ -1,123 +1,152 @@
+  source: VectorSource,
+  projection: any,
+) => {
+  source.clear();
```
Took me awhile to figure this out. The features were disappearing while we loaded a new source when the ids changed. Not the end of the world but I thought we were going to need a loading state. 

This clears the current source, then adds new features so they don't disappear until we pass in new ones.

jimmypalelil commented on public/frontend/declaration.d.ts:
```diff
@@ -1,7 +1,7 @@
 declare module '*.module.css';
 declare module 'happo-playwright';
 declare module 'happo.io';
+declare module 'ol-ext/layer/AnimatedCluster';
```
animations look great 👍

marcellmueller commented:
@jimmypalelil thanks for catching that regression, I fixed the rec resource map page.

As for the clustering [check out this commit](https://github.com/bcgov/nr-rec-resources/pull/986/commits/d97dd32a2bf953fbb8c335f85a815af193f21181), I added `clusterZoomThreshold` option which disables clustering at whatever zoom you give it. I intentionally left the console.log so you can monitor when the clusters are intentionally breaking - at 16 zoom currently.

I think we may still need to handle features that are on the same point or extremely close as this won't stop that overlap from happening.
